### PR TITLE
feat(popover): Simplify popover usage by deprecating popover-manager component

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** N/A

## Summary

feat(popover): Simplify popover usage by deprecating popover-manager component

### CODE REVIEW

**Comment by driskull** [comment]
> To avoid the breaking change, could the popup manager set autoClose on child popups? 

@jcfranco it could, but it would need to setup a mutation observer to do it anytime the dom below it changes otherwise it might not be correct.

IMO I don't think its worth it but let me know.

File: src/components/popover-manager/popover-manager.tsx [hunk 1]
```diff
@@ -36,4 +59,16 @@ export class PopoverManager {
...
     return <slot />;
   }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  setAutoClose = (): void => {
```
Comment (jcfranco) [inline src/components/popover-manager/popover-manager.tsx:69]: Nitpick: This can be a class method instead of an assigned bound function.

**Comment by jcfranco** [comment]
> IMO I don't think its worth it but let me know.

Avoiding a breaking change is totally worth it, IMO.

**Review by jcfranco — APPROVED** [review]
This is awesome stuff, @driskull! 

Approving per our convo (to use a mutation observer). Can you also add a test before merging?

im-so-freakin-excited-shouting

**Pull request merged**